### PR TITLE
Bump QGIS to latest master

### PR DESCRIPTION
**Pull request by m-kuhn**
Preparation for QGIS 3.40

### CODE REVIEW

File: vcpkg/ports/qgis/vcpkg.json [hunk 1]
```diff
@@ -1,138 +1,130 @@
...
+      "dependencies": [
+        "qt5-3d"
+      ]
+    },
+    "bindings": {
+      "description": "Determines whether Python bindings should be built",
+      "dependencies": [
         {
-            "description": "Determines whether the QGIS desktop (main app) should be built"
+          "name": "numpy",
```
Comment (m-kuhn) [inline vcpkg/ports/qgis/vcpkg.json:71]:
```suggestion
          "name": "py-numpy",
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 2]
```diff
@@ -77,7 +77,9 @@ if(MSVC)
   target_compile_definitions(QGIS::Core INTERFACE NOMINMAX)
 endif()
 
-_qgis_core_add_dependency(qca Qca CONFIG)
+_qgis_core_add_dependency(nlohmann_json::nlohmann_json nlohmann_json CONFIG)
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:80]:
```suggestion
_qgis_core_add_dependency(qca Qca CONFIG)
_qgis_core_add_dependency(nlohmann_json::nlohmann_json nlohmann_json CONFIG)
```

File: kadas/gui/search/kadaspinsearchprovider.cpp [hunk 3]
```diff
@@ -71,7 +71,7 @@ void KadasPinSearchProvider::fetchResults( const QString &string, const QgsLocat
 
 void KadasPinSearchProvider::triggerResult( const QgsLocatorResult &result )
 {
-  QVariantMap data = result.getUserData().value<QVariantMap>();
+  QVariantMap data = result.userData().value<QVariantMap>();
```
Comment (m-kuhn) [inline kadas/gui/search/kadaspinsearchprovider.cpp:74]: @3nids @domi4484 

📢 PSA: this is not backwards compatible, once this is merged, dev environments need updating to the latest QGIS master too.

Comment (3nids) [reply kadas/gui/search/kadaspinsearchprovider.cpp:74]: Yes, that'd be great, I'm waiting for a few things in the 3.40 API.

**Comment by m-kuhn** [comment]
@3nids @domi4484  any idea why we have this?

```
RuntimeError: kadas._kadasgui cannot import type 'Qgs3DMapCanvas' from qgis._gui
```

**Comment by 3nids** [comment]
looks like something wrong with the bindings generation. `Qgs3DMapCanvas` should be in `_3d` not in `gui`. You might want to try to  update sipify_all script to integrate 3d. It's not there yet.

But in any case, the sip stuff would require a bit more love, we need to update to the new sipify python to follow QGIS.

**Comment by m-kuhn** [comment]
We don't yet have any 3D in kadas, this is just a bump of QGIS. Possibly QGIS gui python bindings require 3D nowadays?

**Comment by m-kuhn** [comment]
Note to self: I am happy that we introduced this py smoke test, gives already a good first feedback if things work at all

**Comment by m-kuhn** [comment]
@3nids @domi4484 ready for merging

**Comment by 3nids** [comment]
we can merge for me

out of curiosity, this hidden dependency is an issue in Kadas or QGIS?

**Comment by m-kuhn** [comment]
> we can merge for me
> 
> out of curiosity, this hidden dependency is an issue in Kadas or QGIS?

Qgis

**Review by domi4484 — APPROVED** [review]

**Pull request merged**